spec-tests/src/test/java/org/candlepin/spec/HypervisorCheckInSpecTest.java

```diff
@@ -0,0 +1,863 @@
+import org.candlepin.dto.api.client.v1.HypervisorIdDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.dto.api.client.v1.StatusDTO;
+import org.candlepin.dto.api.client.v1.SubscriptionDTO;
+import org.candlepin.dto.api.client.v1.UserDTO;
+import org.candlepin.invoker.client.ApiException;
+import org.candlepin.resource.HostedTestApi;
+import org.candlepin.resource.client.v1.OwnerProductApi;
+import org.candlepin.spec.bootstrap.assertions.OnlyInHosted;
+import org.candlepin.spec.bootstrap.assertions.OnlyInStandalone;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.client.api.ConsumerClient;
+import org.candlepin.spec.bootstrap.client.api.OwnerClient;
+import org.candlepin.spec.bootstrap.data.builder.ConsumerTypes;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.builder.Subscriptions;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.gson.internal.LinkedTreeMap;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@SpecTest
+public class HypervisorCheckInSpecTest {
+    private static ApiClient client;
+    private static OwnerDTO owner;
+    private static OwnerDTO owner2;
+    private static UserDTO user;
+    private OwnerClient ownerApi;
+    private ApiClient userClient;
+    private OwnerProductApi ownerProductApi;
+    private ConsumerClient consumerApi;
+    private HostedTestApi hostedTestApi;
+    private ApiClient hostConsumerClient;
+    private String expectedHostHypervisorId;
+    private String expectedHostName;
+    private List<String> expectedGuestIds;
+    private String uuid1;
+    private String uuid2;
+    private ConsumerDTO hostConsumer;
+    private ConsumerDTO guestConsumer1;
+    private ConsumerDTO guestConsumer2;
+    private ProductDTO virtLimitProduct;
+    private List<PoolDTO> pools;
+    private PoolDTO virtLimitPool;
+    private String hostUuid;
+    private String reporterId;
+
+
+    @BeforeEach
+    public void beforeEach() throws Exception {
```

Comment:
Can you nest inside and already nested test class?

Reply:
Is it necessary to write the tests such that the setup is all-or-nothing? I feel like the repeatedly used bits could probably be moved to a private method or two rather than the global setup.